setup.py

```diff
@@ -42,7 +42,8 @@ def get_long_description() -> str:
         "typed-ast>=1.4.0",
         "regex",
         "pathspec>=0.6, <1",
-    ],
+    ]
+    + (["dataclasses>=0.6"] if sys.version_info < (3, 7) else []),
```

Comment:
Prefer to do it like in https://github.com/python/typing/commit/f254d69c56920819b4211f47fdaba6472097dc5c